src/vision/status_quo/barbara_wants_async_insights.md

```diff
@@ -0,0 +1,70 @@
+
+She really wishes that there was a way to get more insight into why the task is stuck. These were the thoughts inside her head at that moment:
+* Is it waiting on I/O?
+* Is there a deadlock?
+* Did she miss some sync code that might still be there and messing with the executor?
+
+For the I/O question she knows to use some tools on her operating system (lsof). This reveals some open sockets but she's not sure how to act on this.
+
+She scans the code for any std lib imports that might be blocking, but doesn't find anything.
+
+After hours of crawling through the code, she notices that her task is receiving a message from a bounded async channel. She changes this to be an unbounded channel and then things start working.
+
+She wants to know why the code was not working, but unfortunately she has no way to gain insight into this issue. She fears that her task might use too much memory knowing that the channel is unbounded, but she can't really tell.
+
+She thinks, "Anyhow it is working now, let's see if we got some performance gains." After thorough benchmarking she finds out that she didn't quite get the performance gain she was expecting. "Something is not working, as intended", she thinks.
+
+## 🤔 Frequently Asked Questions
+
+*Here are some standard FAQ to get you started. Feel free to add more!*
+
+* **What are the morals of the story?**
+    * There are very few ways to get insights into running systems. Tracing is state of the art. `console.log` #ftw
+    * Tracing is a static activity and there's no way to dynamically gain insights.
+    * While it's possible to find solutions to these issues, often you don't have insight into if those solutions bring new problems.
+    * Debugging process for non-trivial issues is almost guaranteed to be painful and expensive.
+* **What are the sources for this story?**
+    * [Issue 75](https://github.com/rust-lang/wg-async-foundations/issues/75)
+* **Why did you choose [Barbara] to tell this story?**
+    * Barbara knows what she's doing, but still there is little way to get insights, such as:
+        * Custom Events - logging/tracing (Per task?)
+        * Memory consumption per task.
+        * I/O handles in waiting state per task.
+        * Number of tasks and their states over time.
+        * Wake and drop specific tasks.
+        * **Denoised** stack traces and/or stack traces that are task aware.
+        * Who spawned the task?
+        * Worker threads that are blocked from progressing tasks forward.
+        * Tasks that are not progressing.
+* **How would this story have played out differently for the other characters?**
+    * [Alan] who is an Erlang developer, might miss the highly [debuggable](https://youtu.be/JvBT4XBdoUE) BEAM runtime. He would immediately find out that he can't easily analyze a task during runtime.
```

Comment:
Maybe add a note about Java, which also has great tooling, and Go? Many languages have links to good tools for this kind of thing, I think. it'd be nice if we could collect links here, actually. Maybe we can ask for help on this point from The Internet.

Reply:
A list of links:
* [Delve](https://github.com/go-delve/delve), the debugging tool for Go.
* Using [Visual Studio](https://devblogs.microsoft.com/visualstudio/how-do-i-debug-async-code-in-visual-studio/) to debug C#.
* Debugging async Java using [IntelliJ](https://www.jetbrains.com/help/idea/debug-asynchronous-code.html).

Should I insert links like these? Is this what you meant? Correct me if I am wrong.

Reply:
Yep, pefect. I would maybe say "Depending on what languages he was using before, Alan would likely have had experience with a stronger tooling story:" and then list out the various options